### PR TITLE
SN-452 implement custom query for specific entity

### PR DESCRIPTION
**Pull request by toanphamsaritasa**
https://saritasa.atlassian.net/browse/SN-452

### CODE REVIEW

File: src/Saritasa.NetForge.Domain/Entities/Options/EntityOptions.cs [hunk 1]
```diff
@@ -62,4 +62,7 @@ public EntityOptions(Type entityType)
     /// Collection of included navigation names.
     /// </summary>
     public List<string> IncludedNavigations { get; set; } = new();
+
```
Comment (AlexanderMakarov124) [inline src/Saritasa.NetForge.Domain/Entities/Options/EntityOptions.cs:65]: Fix warning here

File: src/Saritasa.NetForge.UseCases/Services/EntityService.cs [hunk 2]
```diff
@@ -224,4 +227,18 @@ public Task<GetEntityByIdDto> GetEntityByIdAsync(string stringId, CancellationTo
 
         return keySelectors;
     }
+
+    private IQueryable<object> ApplyCustomQuery(
+        IQueryable<object> query,
+        Type entityType,
+        ICollection<PropertyMetadataDto> properties,
```
Comment (AlexanderMakarov124) [inline src/Saritasa.NetForge.UseCases/Services/EntityService.cs:234]: `entityType` and `properties` are unused

**Review by AlexanderMakarov124 — APPROVED** [review]

File: src/Saritasa.NetForge.Domain/Entities/Metadata/EntityMetadata.cs [hunk 3]
```diff
@@ -66,4 +66,9 @@ public class EntityMetadata
     /// A collection of navigations metadata associated with this entity.
     /// </summary>
     public List<NavigationMetadata> Navigations { get; set; } = new();
+
+    /// <summary>
+    /// Represents custom query function.
+    /// </summary>
+    public Func<IServiceProvider?, IQueryable<object>, IQueryable<object>>? CustomQuery { get; set; }
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Entities/Metadata/EntityMetadata.cs:73]: `CustomQueryFunction`

File: src/Saritasa.NetForge.Domain/Entities/Options/EntityOptions.cs [hunk 4]
```diff
@@ -62,4 +62,7 @@ public EntityOptions(Type entityType)
     /// Collection of included navigation names.
     /// </summary>
     public List<string> IncludedNavigations { get; set; } = new();
+    
+    /// <inheritdoc cref="EntityMetadata.CustomQuery"/>
+    public Func<IServiceProvider?, IQueryable<object>, IQueryable<object>>? CustomQuery { get; set; }
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Entities/Options/EntityOptions.cs:67]: `CustomQueryFunction`

**Review by vasiliy-chefonov — APPROVED** [review]

**Pull request merged**